src/core/sparsemat.c

```diff
@@ -700,11 +700,18 @@ int igraph_sparsemat_fkeep(igraph_sparsemat_t *A,
                            int (*fkeep)(int, int, igraph_real_t, void*),
                            void *other) {
 
-    if (!cs_fkeep(A->cs, fkeep, other)) {
-        IGRAPH_ERROR("Cannot filter sparse matrix", IGRAPH_FAILURE);
+    if (!A) {
+        IGRAPH_ERROR("The sparse matrix pointer is NULL.", IGRAPH_EINVAL);
     }
+    if (A->cs->nz != -1) {
+        IGRAPH_ERROR("The sparse matrix is not in compressed format.", IGRAPH_EINVAL);
+    }
+    if (!fkeep) {
+        IGRAPH_ERROR("The filter function is a NULL pointer.", IGRAPH_EINVAL);
+    }
+    cs_fkeep(A->cs, fkeep, other);
```

Comment:
It would still be good to check the return value of `cs_fkeep` and see if it is `-1`. Zero should be okay, then all elements were removed. We can link to an external version of this library, which may change in the future.